### PR TITLE
fix(developer): adds utility script to facilitate Developer-build Lerna compatibility

### PR DESCRIPTION
**Pull request by jahorton**
If all goes cleanly, this will replace #3098 as a patch to #3071.

Since the `lexical-model-types` package will no longer have automatic `npm link` commands after transitioning to management with `lerna`, we needed to retool how Developer's compilation links to the package.

We don't want to edit the original, in-repo copy because that would drastically affect development, so it had always been creating an external, temporary copy that could be edited as necessary.  Unfortunately, with it being created outside the repo, `lerna` couldn't help us.  

So, this PR introduces an "intermediate" copy within repo space.  (This copy is under the `windows` build products - `windows\build\intermediate\ModelCompiler`, to be specific - so it won't be checked in.)  A new utility script then facilitates adding and removing this temporary package clone to `lerna` management _just_ long enough to perform the needed `lerna bootstrap` to capture the production dependencies of the module.  This "intermediate" copy is _**then**_ copied to the original temporary location, after which the Makefile proceeds as usual.

### CODE REVIEW

**Comment by jahorton** [comment]
@eddieantonio This is about as far as I can progress on a patch-PR before running into complications with long paths/filenames on Windows.  After discussions with @mcdurdin during standup, we've decided it would be best if you could take the necessary parts for #3071 out of the lmlayer and into their own (private) package/submodule within the repository.

Proposed path / name:
- `common/lexical-model-utils` / `@keymanapp/lexical-model-utils`
   - `@keymanapp/lexical-model-compiler` should depend on this and _**not**_ `@keymanapp/lexical-model-layer`.

This should be enough to prevent a chain-dependency situation causing us to deal with long filenames, at least for Developer compilations, and would allow this to fully proceed.

**Comment by eddieantonio** [comment]
Is this a less hack-y way of packaging things in the developer tools?


> Proposed path / name:
> 
> * `common/lexical-model-utils` / `@keymanapp/lexical-model-utils`
>   
>   * `@keymanapp/lexical-model-compiler` should depend on this and _**not**_ `@keymanapp/lexical-model-layer`.

I can certainly do this. I think the only thing needed is the default word breaker.

File: windows/src/developer/inst/copydev.in [hunk 1]
```diff
@@ -81,16 +83,56 @@ heat-model-compiler:
...
+    npm install
+    # Installs needed dependencies in repo packages (using `npm link` where possible)
+    npm run lerna -- bootstrap -- --production
+
+    # detach from lerna
+    start /wait .\resources\lerna-utils.sh remove-package "windows/bin/intermediate/ModelCompiler"
+    start /wait .\resources\lerna-utils.sh add-package "developer/js"
+!endif
+
+    # node_modules harvest complete - we can now do the rest in true $(TEMP) space.
```
Comment (mcdurdin) [inline windows/src/developer/inst/copydev.in:127]: Can we do all the work in the intermediate folder and not use temp at all now?

**Comment by eddieantonio** [comment]
> I need to test this on my machine to understand the swings and roundabouts in the developer compiler build better.
> 
> > Proposed path / name:
> > 
> > * `common/lexical-model-utils` / `@keymanapp/lexical-model-utils`
> 
> Reading along with #3124, are we better to go with `common/lmlayer/utils` and `common/lmlayer/types` and `common/lmlayer/prediction` (instead of `common/predictive-text`)?

Oooh! I like the sound of `common/lm/types`, `common/lm/models`, `common/lm/word-breakers`! I'll keep this in mind in future PRs!

File: windows/src/developer/inst/copydev.in (hunk 1 above)
Comment (jahorton) [reply windows/src/developer/inst/copydev.in:127]: We could probably move some of the `rmdir` commands there, to be sure... but I'm fairly certain the 'wix' setup is dependent on the final file path, so we'd still want a final `xcopy` to the original location.  That's what my interpretation of [these lines](https://github.com/keymanapp/keyman/blob/master/windows/src/developer/inst/copydev.in#L13-L18) tells me, anyway.

Of course, I'm also less comfortable with makefiles, let alone the implications of the `.in` filetype (I see that the `.in` gets compiled into a `.mak`?) and the complicated makefile hierarchy of the Developer project as a whole, so I figured I'd leave as much of the original process in place as I could - less chances that I might break something.

**Comment by jahorton** [comment]
> > I need to test this on my machine to understand the swings and roundabouts in the developer compiler build better.
> > > Proposed path / name:
> > > 
> > > * `common/lexical-model-utils` / `@keymanapp/lexical-model-utils`
> > 
> > 
> > Reading along with #3124, are we better to go with `common/lmlayer/utils` and `common/lmlayer/types` and `common/lmlayer/prediction` (instead of `common/predictive-text`)?
> 
> Oooh! I like the sound of `common/lm/types`, `common/lm/models`, `common/lm/word-breakers`! I'll keep this in mind in future PRs!

I'm partial to the `common/lmlayer/*` variants myself, and found myself wanting to implement them anyway as I worked through this PR's issues.  Glad we all agree about the general reorganization idea!

**Comment by mcdurdin** [comment]
Local test failed for me.

```
C:\projects\keyman\app\windows\bin\intermediate\ModelCompiler\node_modules\@keymanapp\lexical-model-layer\node_modules\@keymanapp\keyboard-processor\node_modules\@keymanapp\lexical-model-types\node_modules\typescript\lib\cancellationToken.js
Insufficient memory
186 File(s) copied

** error 4 ** deleting heat-model-compiler

** error 1 ** deleting setup

** error 1 ** deleting build-release-wrapper

** error 1 ** deleting release
```

Given this error, will wait until we have had our namespacing discussion before continuing review.

**Comment by jahorton** [comment]
Closing in favor of #3166, which is far cleaner.

**Pull request closed without merging**